### PR TITLE
Extract AbstractAuthenticationService to share logic between jira and confluence

### PR DESCRIPTION
**Pull request by pathob**
The shared service plumbing (bulk set with declarative null-model and map-key fallback semantics, single-entity set, find-by-name, get/set SSO) lives in a new AbstractAuthenticationService in commons. Host-specific subclasses provide abstract hooks that delegate to their existing per-host model utils.

Add tests for the null-model and null-name semantics in both AuthenticationServiceTests.

### CODE REVIEW

File: commons/src/main/java/com/deftdevs/bootstrapi/commons/service/AbstractAuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                continue;
+            }
+
+            if (idpModel.getName() == null) {
+                idpModel.setName(idpName);
+            }
+
+            final IB resultIdpModel = setAuthenticationIdp(idpModel);
+            resultIdpModels.put(resultIdpModel.getName(), resultIdpModel);
+        }
```
[the remark this answers is not in the document]
Comment (pathob) [reply commons/src/main/java/com/deftdevs/bootstrapi/commons/service/AbstractAuthenticationService.java:64]: @copilot apply changes based on [this feedback](https://github.com/deftdevs/bootstrapi/pull/477#discussion_r3235060779)
